src/physics/land.jl

```diff
@@ -27,7 +27,7 @@ Base.@kwdef struct SeasonalLandTemperature{NF, Grid<:AbstractGrid{NF}} <: Abstra
     varname::String = "lst"
 
     "Grid the land surface temperature file comes on"
-    file_Grid::Type{<:AbstractGrid} = FullGaussianGrid
+    file_Grid::Type{<:AbstractGrid} = FullGaussianGrid{NF}
```

Comment:
Why was that necessary? The idea here is to just store the information of what grid the gridded netcdf file comes on -- technically one could infer this from the latitudes, but I thought that was tedious

Reply:
Yes you are right, I'll remove it again. 
Initially I had a version of this PR that was trying to do this solely by multiple dispatch. But this wasn't working nicely

Reply:
There is a related issue that NCDatasets returns `Union{Missing, Float32}` as eltype maybe when reading the netcdf file this was dealt with by doing `FullGaussianGrid{NF}(...) ` so fixing that type on the side of the grid instead of doing something like `FullGaussianGrid(Array{NF}(ds[var]))` which maybe we should do generally?

Reply:
Yes, I am aware that NCDatasets returns `Union{Missing, Float32}` but where do we read like that? I've only seen something like `file_Grid(ncfile["orog"][:, :])` in the code, which seems to work without the `NF` (why actually?).

Reply:
I think it works because I didn't specify `_FillValue` for those netcdf files in which case the returned eltype is Float32 and not a Union. In any case we maybe should do `Grid(Array(ds["varname"].var), input_as=Matrix)` which always ignores masked values, doesn't require the dimensions and obtains the eltype from the netcdf automatically?

Reply:
Does this always work? I just tried it with some netCDF file I had lying around. If the netCDF variables is `Union{Missing, Float64}`, 
* `Array(ds["varname"])` also results in an `Array{Union{Missing, Float64}, 3}` 
* and `Array(ds["varname"].var)` in `Array{Float32, 3}`
I find this a bit confusing, but if you're sure it always works, it's probably safer to change it to ``Array(ds["varname"].var)`

Reply:
Did some speed comparison https://github.com/Alexander-Barth/NCDatasets.jl/issues/227#issuecomment-2321498232 I believe in our case we should do one of these

1. No missing values (i.e. ignore them), dimensions known, number format as in NetCDF
```julia
ds["vor"].var[:, :, :, :]
```

2. No missing values (i.e. ignore them), number format as in NetCDF but dimensions unknown

```julia
Array(ds["vor"].var)
```

3. like 2 but convert to another number format on the fly

```julia
Array(ds["vor"].var)
```

Reply:
Implemented 1. with [e81208c](https://github.com/SpeedyWeather/SpeedyWeather.jl/pull/572/commits/e81208c06be4db11e7e15f26cac61c2a8c738492)